fineract-provider/src/main/java/org/apache/fineract/portfolio/delinquency/service/DelinquencyReadPlatformServiceImpl.java

```diff
@@ -97,4 +107,88 @@ public Collection<LoanDelinquencyTagHistoryData> retrieveDelinquencyRangeHistory
         return mapperLoanDelinquencyTagHistory.map(loanDelinquencyTagData);
     }
 
+    @Override
+    public CollectionData calculateLoanCollectionData(final Long loanId) {
+        final LocalDate businessDate = DateUtils.getBusinessLocalDate();
+        final Loan loan = this.loanRepository.getReferenceById(loanId);
```

Comment:
Here we need more than the reference. (it not cause any issues, but still...feel free to fetch the loan by findById..)

Reply:
Done